Sources/TokamakDOM/Views/List.swift

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TokamakCore
+
+public typealias List = TokamakCore.List
+
+extension List: ViewDeferredToRenderer {
+  public var deferredBody: AnyView {
+    AnyView(HTML("ul", [
+      "class": "_tokamak-list",
+    ]) { () -> AnyView in
+      let content = _ListProxy(self).content
+      if let parentView = content as? ParentView {
+        print(parentView.children.count)
+        return AnyView(ForEach(Array(parentView.children.enumerated()), id: \.offset) { _, view in
+          HTML("li") {
+            VStack {
+              view
+              Divider()
+            }
+          }
+        })
```

Comment:
Ideally this would do some windowing so a list with thousands of elements wouldn’t be terribly slow and DOM-intensive but that would require measuring views. A quick test shows SwiftUI renders the first 20 (15 on macOS) views in a List to measure them, then throws away the result for the ones that are off-screen. This leads to recalculations of the actual scroll height if the cells are not identically sized.

A `List { ForEach { ... } }` renders the first 20/15 views but doesn’t re-render them.

Reply:
Totally fine to punt this to an issue though

Reply:
hm, I thought that lazy loading of views is something that only new views like [LazyVStack](https://developer.apple.com/documentation/swiftui/lazyvstack) do.

Reply:
Nope! I tested it out with this snippet:

<details>

```swift
struct ContentView: View {
    var body: some View {
      List(1..<1000) { i in
        RowView(row: i)
      }
    }
}

struct RowView: View {
  @State var height = Int.random(in: 1..<10) * 100
  var row: Int
  var body: some View {
    print("Evaling \(row)")
    return Text("Row \(row)")
      .frame(height: CGFloat(height), alignment: .center)
  }
}
```

Reply:
One downside of lazy loading of `List` content in DOM could be that if you do a page search, it will only show the currently loaded elements in search results. I agree that there's no need to implement it in this PR, and maybe we can make lazy loading for `List`s configurable later, if we implement it at all. Maybe we could just redirect people to `LazyVStack` in the meantime, after we get that one ready 🙂

Reply:
Using the Intersection Observer API, this could be fairly simple.

1. Create the first n children (to fill at least part of the screen) and setup an `IntersectionObserver` on each one.
2. When an element intersects with the root element (with a margin), create the next several items if they are not already visible.

Configurable definitely sounds preferable, but I think lazy loading could be default as its the closest match to SwiftUI. Pagination could even be an option.